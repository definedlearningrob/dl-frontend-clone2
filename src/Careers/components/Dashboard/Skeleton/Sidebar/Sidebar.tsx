import { DCLogo } from '@dc/shared/DCLogo/DCLogo';

import SkeletonSidebarFooter from './Footer/Footer';
import SkeletonSidebarItem from './Item/Item';

type Props = {
  itemsCount: number;
};

function SkeletonSidebar({ itemsCount }: Props) {
  const renderItems = () =>
    Array.from({ length: itemsCount }, (_, index) => <SkeletonSidebarItem key={index} />);

  return (
    <div className='skeleton-sidebar'>
      <DCLogo className='skeleton-sidebar__logo-icon' />
      {renderItems()}
      <div className='skeleton-sidebar__divider' />
      <SkeletonSidebarFooter />
    </div>
  );
}

export default SkeletonSidebar;
