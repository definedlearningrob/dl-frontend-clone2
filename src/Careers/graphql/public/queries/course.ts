import { gql, TypedDocumentNode } from '@apollo/client';

import { EducationalSettingTypes } from '@dc/resources/enums';

export const PUBLIC_COURSE_QUERY: TypedDocumentNode<TCourseData, TCourseVariables> = gql`
  query PublicCourse($shareId: ID!, $code: String!, $track: Boolean) {
    course(shareId: $shareId, code: $code, track: $track) {
      id
      description
      name
      pathway {
        name
      }
      lessons {
        name
        description {
          audience
          introduction
          goal
          role
          situation
        }
        archivedAt
        assignments {
          description
          displayName
          id
          rubrics {
            criteriaLabels {
              displayName
              id
              score
              uuid
            }
            criterias {
              id
              rubricCriteriaLabelId
              rubricHeadingId
              text
              uuid
            }
            description
            headings {
              id
              multiplier
              name
              uuid
            }
            id
            name
            uuid
          }
          step
        }
        attachments {
          description
          displayName
          files {
            filename
            id
            url
          }
          id
          step
        }
        careerReviewSurvey {
          questions {
            id
          }
        }
        checkInGroups {
          displayName
          id
          name
          questions {
            id
            question
            step
          }
          step
        }
        checkInQuestions {
          id
          question
          step
        }
        externalPresentations {
          displayName
          isExpandable
          id
          source
          step
        }
        hasPresentation
        id
        imageUrl
        name
        researchLinks {
          author
          displayName
          id
          resourceLink
          sourceName
          step
        }
        step
        texts {
          content
          displayName
          id
          step
        }
        thumbnailUrl
        type
        videos {
          description
          displayName
          filename
          id
          step
          url
        }
        vocabularies {
          definition
          id
          step
          term
        }
      }
    }
  }
`;

export type TCourseLessonAssignmentRubric = {
  criteriaLabels: {
    displayName: string;
    id: string;
    score: number;
    uuid: string;
  };
  criterias: {
    id: string;
    rubricCriteriaLabelId: string;
    rubricHeadingId: string;
    text: string;
    uuid: string;
  };
  description: string;
  headings: {
    id: string;
    multiplier: number;
    name: string;
    uuid: string;
  };
  id: string;
  name: string;
};

export type TCourseLessonAssignment = {
  description: string;
  displayName: string;
  id: string;
  rubrics: TCourseLessonAssignmentRubric;
  step: number;
};

export type TCourseLessonAttachment = {
  description: string;
  displayName: string;
  files: {
    filename: string;
    id: string;
    url: string;
  };
  id: string;
  step: number;
};

export type TCourseLessonCareerReviewSurvey = {
  questions: {
    id: string;
    options: {
      option: string;
      step: number;
    };
    question: string;
    type: string;
  };
};

export type TCourseLessonCheckInGroup = {
  displayName: string;
  id: string;
  name: string;
  questions: {
    id: string;
  };
  step: number;
};

export type TCourseLessonCheckInQuestion = {
  id: string;
  question: string;
  step: number;
};

export type TCourseLessonDescription = {
  audience: string;
  goal: string;
  introduction: string;
  role: string;
  situation: string;
};

export type TCourseLessonExternalPresentation = {
  displayName: string;
  description: string;
  id: string;
  isExpandable: boolean;
  source: string;
  step: number;
};

export type TCourseLessonResearchLink = {
  author: string;
  displayName: string;
  id: string;
  resourceLink: string;
  sourceName: string;
  step: number;
};

export type TCourseLessonText = {
  content: string;
  displayName: string;
  id: string;
  step: number;
};

export type TCourseLessonVideo = {
  description: string;
  displayName: string;
  filename: string;
  id: string;
  step: number;
  url: string;
};

export type TCourseLessonVocabulary = {
  definition: string;
  id: string;
  step: number;
  term: string;
};

export type TPublicCourseLesson = {
  archivedAt: string;
  assignments: TCourseLessonAssignment[];
  attachments: TCourseLessonAttachment[];
  careerReviewSurvey: TCourseLessonCareerReviewSurvey;
  checkInGroups: TCourseLessonCheckInGroup[];
  checkInQuestions: TCourseLessonCheckInQuestion[];
  description?: TCourseLessonDescription;
  externalPresentations: TCourseLessonExternalPresentation[];
  hasPresentation?: boolean;
  id: string;
  imageUrl: string;
  name: string;
  researchLinks: TCourseLessonResearchLink[];
  step: number;
  texts: TCourseLessonText[];
  thumbnailUrl: string;
  type: string;
  videos: TCourseLessonVideo[];
  vocabularies: TCourseLessonVocabulary[];
};

export type TCoursePathway = {
  id: string;
  name: string;
};

export type TCourse = {
  description: string;
  id: string;
  imageUrl: string;
  thumbnailUrl: string;
  lesson: TPublicCourseLesson;
  lessons: TPublicCourseLesson[];
  name: string;
  pathway: TCoursePathway;
  type: EducationalSettingTypes;
};

export type TCourseVariables = {
  id?: string;
  code: string;
  track?: boolean;
  shareId: string;
};

export type TCourseData = {
  course: TCourse;
};
