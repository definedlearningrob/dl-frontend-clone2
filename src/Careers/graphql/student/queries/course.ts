import { gql } from '@apollo/client';

import { EducationalSettingTypes } from '@dc/resources/enums';

export default gql`
  query StudentCourse($id: ID!, $track: Boolean) {
    course(id: $id, track: $track) {
      description
      id
      imageUrl
      lessons {
        careerReviewSurvey {
          performed
          version
        }
        id
        imageUrl
        name
        progress {
          submitted
          total
        }
        step
        type
        thumbnailUrl
      }
      name
      pathway {
        name
      }
      progress {
        submitted
        total
      }
      type
    }
  }
`;

export type TLessonProgress = {
  submitted: number;
  total: number;
};

export type TLessonCareerReviewSurvey = {
  performed: boolean;
};

export type TCourseLesson = {
  careerReviewSurvey: TLessonCareerReviewSurvey;
  id: string;
  imageUrl: string;
  name: string;
  progress: TLessonProgress;
  step: number;
  type: string;
  thumbnailUrl: string;
};

export type TCourseProgress = {
  submitted: number;
  total: number;
};

export type TCoursePathway = {
  name: string;
};

export type TCourse = {
  description: string;
  id: string;
  imageUrl: string;
  lessons: TCourseLesson[];
  name: string;
  pathway: TCoursePathway;
  progress: TCourseProgress;
  type: EducationalSettingTypes;
};

export type TCourseVariables = {
  id: string;
  track: boolean;
};

export type TCourseData = {
  course: TCourse;
};
