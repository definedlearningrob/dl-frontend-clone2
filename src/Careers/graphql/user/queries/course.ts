import { gql } from '@apollo/client';

import courseMetadataFragment from '@dc/graphql/fragments/courseMetadata';
import { EducationalSettingTypes } from '@dc/resources/enums';
import { TSharedResource } from '@dc/graphql/user/mutations/shareResource';
import { TCollection } from '@dc/resources/types';

import { ContentStatusesTypes } from '@shared/resources/enums';

export default gql`
  query UserCourse($id: ID!, $track: Boolean) {
    course(id: $id, track: $track) {
      description
      displayName
      badges {
        id
        imageUrl
        name
      }
      id
      imageUrl
      isGlobal
      thumbnailUrl
      lessons {
        id
        imageUrl
        name
        step
        thumbnailUrl
        type
      }
      name
      sharedResource {
        allowLogin
        code
      }
      status
      pathway {
        id
        name
      }
      type
      collection {
        id
        name
      }
      ...CourseMetadata
    }
  }
  ${courseMetadataFragment}
`;

export type TCourseLesson = {
  id: string;
  imageUrl: string;
  name: string;
  step: number;
  thumbnailUrl: string;
  type: string;
};

export type TCoursePathway = {
  id: string;
  name: string;
};

export type TCourse = {
  description: string;
  displayName: string;
  id: string;
  imageUrl: string;
  isGlobal: boolean;
  thumbnailUrl: string;
  lessons: TCourseLesson[];
  name: string;
  status: ContentStatusesTypes;
  pathway: TCoursePathway;
  type: EducationalSettingTypes;
  sharedResource: TSharedResource;
  collection: TCollection;
};

export type TCourseVariables = {
  id: string;
  track: boolean;
};

export type TCourseData = {
  course: TCourse;
};
