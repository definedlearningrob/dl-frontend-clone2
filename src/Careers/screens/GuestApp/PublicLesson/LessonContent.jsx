import { useMemo } from 'react';
import PropTypes from 'prop-types';

import Description from '@dc/components/Student/Lesson/Description/Description';
import LessonExtensionFields from '@dc/components/Extensions/Lessons/DeepDive/ExtensionFields';
import Assignment from '@dc/components/Student/Lesson/Assignment/Assignment';
import Attachment from '@dc/components/Student/Lesson/Attachment/Attachment';
import Presentation from '@dc/components/Student/Lesson/ExternalPresentation/ExternalPresentation';
import ResearchLink from '@dc/components/Student/Lesson/ResearchLink/ResearchLink';
import Text from '@dc/components/Student/Lesson/Text/Text';
import Video from '@dc/components/Student/Lesson/Video/Video';
import Vocabulary from '@dc/components/Student/Lesson/Vocabulary/Vocabulary';
import CheckInQuestion from '@dc/components/Student/Lesson/CheckInQuestion/CheckInQuestion';
import CheckInGroup from '@dc/components/Student/Lesson/CheckInGroup/CheckInGroup';
import { LESSON_TYPES } from '@dc/resources/constants';

export const LessonContent = ({ lesson, isPublic }) => {
  const sortedLessonItems = useMemo(() => {
    const {
      externalPresentations,
      vocabularies,
      texts,
      attachments,
      videos,
      assignments,
      researchLinks,
    } = lesson;

    return [
      ...externalPresentations,
      ...vocabularies,
      ...texts,
      ...attachments,
      ...videos,
      ...assignments,
      ...researchLinks,
    ].sort((a, b) => a.step - b.step);
  }, [lesson]);

  const groupedItems = useMemo(() => {
    const itemsThatShouldBeGroupes = ['researchlink', 'vocabulary'];

    return sortedLessonItems.reduce((acc, item, index, array) => {
      const nextItem = array[index + 1];
      const previousItem = array[index - 1];
      const previousItemType = previousItem && previousItem.__typename.toLowerCase();
      const nextItemType = nextItem && nextItem.__typename.toLowerCase();
      const itemType = item.__typename.toLowerCase();
      const itemIsGroupable = itemsThatShouldBeGroupes.includes(itemType);
      const nextItemIsSameType = nextItemType === itemType;
      const previousItemIsSameType = previousItemType === itemType;

      if (itemIsGroupable && nextItemIsSameType && !previousItemIsSameType) {
        return [...acc, { __typename: item.__typename, items: [item] }];
      }

      if (itemIsGroupable && previousItemIsSameType) {
        acc[acc.length - 1].items.push(item);

        return acc;
      }

      return [...acc, item];
    }, []);
  }, [sortedLessonItems]);

  const lessonItemCards = useMemo(
    () =>
      groupedItems.map(
        (item) =>
          ({
            assignment: (
              <Assignment key={item.step + item.id} assignment={item} previewOnly={true} />
            ),
            attachment: (
              <Attachment key={item.step + item.id} attachment={item} previewOnly={true} />
            ),
            externalpresentation: <Presentation key={item.step + item.id} presentation={item} />,
            researchlink: (
              <ResearchLink
                key={
                  (item.items ? item.items[0].step : item.step) +
                  (item.items ? item.items[0].id : item.id)
                }
                previewOnly={true}
                researchLinks={item.items || [item]}
              />
            ),
            text: <Text key={item.step + item.id} text={item} />,
            video: <Video key={item.step + item.id} video={item} />,
            vocabulary: (
              <Vocabulary
                key={
                  (item.items ? item.items[0].step : item.step) +
                  (item.items ? item.items[0].id : item.id)
                }
                vocabularies={item.items || [item]}
              />
            ),
          }[item.__typename.toLowerCase()])
      ),
    [groupedItems]
  );

  const renderCheckins = () => {
    const { checkInQuestions, checkInGroups } = lesson;

    const checkins = [...checkInGroups, ...checkInQuestions];

    const sortedCheckins = checkins.sort(
      (a, b) => parseInt(String(a.step)) - parseInt(String(b.step))
    );

    return sortedCheckins.map(
      (checkin) =>
        ({
          checkinquestion: (
            <CheckInQuestion
              key={`check-in-${checkin.question}`}
              checkInQuestion={checkin}
              previewOnly={true}
            />
          ),
          checkingroup: (
            <CheckInGroup
              key={`check-in-group-${checkin.displayName}`}
              checkInGroup={checkin}
              previewOnly={true}
            />
          ),
        }[checkin.__typename.toLowerCase()])
    );
  };

  const presentationCards = useMemo(
    () => lessonItemCards?.filter((item) => item.props.presentation),
    [lessonItemCards]
  );
  const withoutPresentationCards = useMemo(
    () => lessonItemCards.filter((item) => !item.props.presentation),
    [lessonItemCards]
  );

  const isLessonTypeOfDigDeeper = useMemo(
    () => lesson?.type === LESSON_TYPES.DIG_DEEPER_INTO_CAREER.toLowerCase(),
    [lesson]
  );

  return (
    <div>
      <section className='lesson'>
        <h2 className='lesson__title'>{lesson?.name}</h2>
        <section className='presentation'>{presentationCards}</section>
        <Description lesson={lesson} />
        {isLessonTypeOfDigDeeper && !isPublic && <LessonExtensionFields />}
        {withoutPresentationCards}
        {renderCheckins()}
      </section>
    </div>
  );
};

LessonContent.propTypes = {
  isPublic: PropTypes.bool,
  lesson: PropTypes.shape({
    assignments: PropTypes.array,
    attachments: PropTypes.array,
    careerReviewSurvey: PropTypes.object,
    checkInGroups: PropTypes.array,
    checkInQuestions: PropTypes.array,
    externalPresentations: PropTypes.array,
    id: PropTypes.string,
    name: PropTypes.string,
    researchLinks: PropTypes.array,
    texts: PropTypes.array,
    type: PropTypes.string,
    videos: PropTypes.array,
    vocabularies: PropTypes.array,
  }),
};
