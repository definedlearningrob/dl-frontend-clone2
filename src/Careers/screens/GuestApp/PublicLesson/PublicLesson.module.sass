@import '@shared/stylesheets/config.sass'

.container
  margin-top: 0 !important
