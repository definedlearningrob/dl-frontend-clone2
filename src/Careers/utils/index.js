const isObject = (object) => object !== null && typeof object === 'object';

export const isEqual = (object1, object2) => {
  if (object1 === object2) return true;

  if (
    object1 === null ||
    object2 === null ||
    typeof object1 !== 'object' ||
    typeof object2 !== 'object'
  )
    return false;

  const keys1 = Object.keys(object1);
  const keys2 = Object.keys(object2);

  if (keys1.length !== keys2.length) {
    return false;
  }

  for (let key of keys1) {
    const val1 = object1[key];
    const val2 = object2[key];
    const areObjects = isObject(val1) && isObject(val2);
    if (
      // eslint-disable-next-line
      areObjects && !isEqual(val1, val2) ||
      // eslint-disable-next-line
      !areObjects && val1 !== val2
    ) {
      return false;
    }
  }

  return true;
};

export const getByStringKey = (object, key) => {
  const keys = key.split('.');

  return keys.reduce((acc = {}, key) => acc[key], object);
};
