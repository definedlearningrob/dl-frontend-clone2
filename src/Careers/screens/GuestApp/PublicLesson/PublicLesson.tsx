import { useParams } from 'react-router-dom';
import { useEffect } from 'react';

import SharedMainContent from '@dc/shared/MainContent/MainContent';
import { usePublicCourseQuery } from '@dc/graphql/public/hooks/usePublicCourseQuery';
import { LessonContent } from '@dc/screens/GuestApp/PublicLesson/LessonContent';

import SharedLoadingSpinner from '@shared/components/LoadingSpinner/LoadingSpinner';
import useQueryParams from '@shared/hooks/useQueryParams';
import { useNavigation } from '@shared/components/Sidebar/useNavigation';

import styles from './PublicLesson.module.sass';

export const PublicLesson = () => {
  const { lessonId, shareId } = useParams<{ lessonId: string; shareId: string }>();
  const { setBackNavButton } = useNavigation();
  const {
    params: { code },
  } = useQueryParams<{ code: string }>();

  const { data, loading } = usePublicCourseQuery({
    code,
    shareId,
  });

  const foundLesson = data?.course?.lessons?.find(
    (lesson: { id: string }) => lesson.id === lessonId
  );

  useEffect(() => {
    setBackNavButton(true);

    return () => {
      setBackNavButton(false);
    };
  }, []);

  if (loading) {
    return (
      <SharedMainContent>
        <SharedLoadingSpinner size='full-screen' />
      </SharedMainContent>
    );
  }

  if (!data) {
    return null;
  }

  return (
    <SharedMainContent className={styles.container}>
      {foundLesson ? (
        <LessonContent isPublic={true} lesson={foundLesson} />
      ) : (
        <SharedLoadingSpinner size='full-screen' />
      )}
    </SharedMainContent>
  );
};
