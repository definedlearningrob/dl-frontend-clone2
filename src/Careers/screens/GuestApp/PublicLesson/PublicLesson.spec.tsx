import { MockedProvider } from '@apollo/client/testing';
import { Redirect, Route, Switch } from 'react-router-dom';
import { screen } from '@testing-library/dom';

import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import { PublicLesson } from '@dc/screens/GuestApp/PublicLesson/PublicLesson';
import { PUBLIC_COURSE_QUERY } from '@dc/graphql/public/queries/course';

import { NavigationContextProvider } from '@shared/components/Sidebar/useNavigation';

const courseQuerySpy = jest.fn();

const mocks = [
  {
    request: {
      query: PUBLIC_COURSE_QUERY,
      variables: {
        code: undefined,
        shareId: undefined,
        track: true,
      },
    },
    result: () => {
      courseQuerySpy();

      return {
        data: {
          course: {
            id: '1',
            description: 'Course 1 description',
            name: 'Course 1',
            pathway: {
              name: 'Pathway 1',
            },
            lessons: [
              {
                archivedAt: null,
                assignments: [],
                attachments: [],
                careerReviewSurvey: null,
                checkInGroups: [
                  {
                    displayName: 'Lesson 1 Check-ins',
                    id: '1',
                    name: 'Career cluster',
                    questions: [
                      {
                        id: '1',
                        question:
                          'What do you like best about the careers and work done within this career cluster?',
                        step: 1,
                        __typename: 'CheckInQuestion',
                      },
                    ],
                    step: 1,
                    __typename: 'CheckInGroup',
                  },
                ],
                checkInQuestions: [],
                description: {
                  __typename: 'LessonDescription',
                  audience: '',
                  introduction: '',
                  situation: '',
                  goal: '',
                  role: '',
                },
                externalPresentations: [
                  {
                    displayName: 'Overview',
                    id: '14',
                    source: 'https://slides.definedlearning.com/p/c9',
                    step: 1,
                    __typename: 'ExternalPresentation',
                    isExpandable: false,
                  },
                ],
                hasPresentation: true,
                id: '2',
                imageUrl:
                  'http://localstack.lvh.me:4566/dev-bucket/images/lessons/dd057967-4ca9-4bc9-91b8-ea59a294838b/manufacturing.jpg?response-cache-control=max-age%3D3600&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=foobar%2F20230524%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230524T062019Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=0dfbe59a58371d4234ef4ee019d6c451e5c0e9d99c9362a1fc88b89097d93257',
                name: 'Career Cluster: Manufacturing',
                researchLinks: [],
                step: 1,
                texts: [],
                thumbnailUrl:
                  'http://localstack.lvh.me:4566/dev-bucket/images/lessons/dd057967-4ca9-4bc9-91b8-ea59a294838b/manufacturing_thumbnail.png?response-cache-control=max-age%3D3600&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=foobar%2F20230524%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230524T062019Z&X-Amz-Expires=3600&X-Amz-SignedHeaders=host&X-Amz-Signature=77978a4f95b41e80897566eaef12002c6ba671306bad02455b5803d27d9510a7',
                type: 'career_cluster',
                videos: [],
                vocabularies: [],
                __typename: 'Lesson',
              },
            ],
          },
        },
      };
    },
  },
];

const renderLesson = () =>
  renderWithRouterAndReduxProvider(
    <MockedProvider mocks={mocks}>
      <NavigationContextProvider>
        <Switch>
          <Route path='shared/student/courses/:courseId/lessons/:lessonId'>
            <PublicLesson />
          </Route>
          <Redirect from='*' to='/sign-in' />
        </Switch>
      </NavigationContextProvider>
    </MockedProvider>,
    { route: 'shared/student/courses/1/lessons/2' }
  );

describe('Lesson', () => {
  it('renders main lesson content', async () => {
    const { container } = renderLesson();
    const title = await screen.findByText('Career Cluster: Manufacturing');
    const checkInGroupTitle = await screen.findByText('Lesson 1 Check-ins');
    const checkInGroupQuestion = await screen.findByText(
      'What do you like best about the careers and work done within this career cluster?'
    );
    const buttons = await screen.findAllByRole('button', { name: 'Save' });
    const textArea = await screen.findAllByRole('textbox');

    expect(title).toBeInTheDocument();
    expect(checkInGroupTitle).toBeInTheDocument();
    expect(checkInGroupQuestion).toBeInTheDocument();
    expect(buttons.length).toBe(1);
    expect(textArea.length).toBe(1);
    expect(buttons[0]).toBeDisabled();
    expect(textArea[0]).toBeDisabled();
    expect(container).toMatchSnapshot();
  });
});
