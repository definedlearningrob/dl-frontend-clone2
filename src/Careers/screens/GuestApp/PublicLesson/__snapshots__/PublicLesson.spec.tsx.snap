// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lesson renders main lesson content 1`] = `
<div>
  <div
    class="p-base pt-xs xxxl:p-md xxxl:pt-sm container"
    id="read"
  >
    <div>
      <section
        class="lesson"
      >
        <h2
          class="lesson__title"
        >
          Career Cluster: Manufacturing
        </h2>
        <section
          class="presentation"
        >
          <div
            class="lesson-item-card !p-0 !p-base"
            data-testid="lesson-item-presentation"
            id="ExternalPresentation14"
          >
            <div
              class="lesson-item-card__header"
            >
              <h4
                class="lesson-item-card__title"
              >
                Overview
              </h4>
            </div>
            <div
              class="h-[420px] w-full transition-all relative"
            >
              <div
                class="absolute inset-0 hidden"
              />
              <iframe
                allowfullscreen=""
                class="rounded-sm w-full h-full border-0"
                sandbox="allow-same-origin allow-scripts allow-popups allow-forms"
                src="https://slides.definedlearning.com/p/c9"
                title="video"
              />
            </div>
          </div>
        </section>
        <div
          class="lesson-item-card student-check-in-question"
          data-testid="lesson-item-check-in-question"
          id="CheckInGroup1"
        >
          <div
            class="lesson-item-card__header"
          >
            <h4
              class="lesson-item-card__title"
            >
              Lesson 1 Check-ins
            </h4>
          </div>
          <div
            class="student-check-in-question__item"
          >
            <h4
              class="student-check-in-question__title"
            >
              What do you like best about the careers and work done within this career cluster?
            </h4>
            <div
              class="flex flex-col gap-xs student-check-in-question__textarea mb-sm"
            >
              <label
                class="flex flex-col gap-xs"
              >
                <textarea
                  class="textarea__input !mb-0"
                  disabled=""
                  name="answer"
                  placeholder="Type here..."
                />
              </label>
            </div>
            <div
              class="flex flex-row content-center justify-center"
            >
              <button
                class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] ml-auto student-check-in-question__save-button -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-base leading-[22px] border text-white"
                data-testid="button"
                disabled=""
                type="submit"
              >
                <span
                  class="items-center flex justify-center gap-xs"
                >
                  Save
                </span>
              </button>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
